doc/application/coccinelle.rst

```diff
@@ -0,0 +1,510 @@
+
+http://coccinelle.lip6.fr/documentation.php
+
+Using Coccinelle on Zephyr
+**************************
+
+``coccicheck`` checker is the front-end to the Coccinelle infrastructure
+and has various modes:
+
+Four basic modes are defined: ``patch``, ``report``, ``context``, and
+``org``. The mode to use is specified by setting the MODE variable with
+``--mode=<mode>`` or ``-m=<mode>``.
+
+* ``patch`` proposes a fix, when possible.
+
+* ``report`` generates a list in the following format:
+  file:line:column-column: message
+
+* ``context`` highlights lines of interest and their context in a
+  diff-like style.Lines of interest are indicated with ``-``.
+
+* ``org`` generates a report in the Org mode format of Emacs.
+
+Note that not all semantic patches implement all modes. For easy use
+of Coccinelle, the default mode is ``report``.
+
+Two other modes provide some common combinations of these modes.
+
+- ``chain`` tries the previous modes in the order above until one succeeds.
+
+- ``rep+ctxt`` runs successively the report mode and the context mode.
+  It should be used with the C option (described later)
+  which checks the code on a file basis.
+
+Examples
+********
+
+To make a report for every semantic patch, run the following command:
+
+.. code-block:: console
+
+        ./scripts/coccicheck --mode=report
+
+To produce patches, run:
+
+.. code-block:: console
+
+        ./scripts/coccicheck --mode=patch
+
+The coccicheck target applies every semantic patch available in the
+sub-directories of ``scripts/coccinelle`` to the entire source code tree.
+
+For each semantic patch, a commit message is proposed.  It gives a
+description of the problem being checked by the semantic patch, and
+includes a reference to Coccinelle.
+
+As any static code analyzer, Coccinelle produces false
+positives. Thus, reports must be carefully checked, and patches
+reviewed.
+
+To enable verbose messages set --verbose=1 option, for example:
+
+.. code-block:: console
+
+        ./scripts/coccicheck --mode=report --verbose=1
+
+Coccinelle parallelization
+**************************
+
+By default, ``coccicheck`` tries to run as parallel as possible. To change
+the parallelism, set the --jobs=<number> option. For example, to run across
+4 CPUs:
+
+.. code-block:: console
+
+        ./scripts/coccicheck --mode=report --jobs=4
+
+As of Coccinelle 1.0.2 Coccinelle uses Ocaml parmap for parallelization,
+if support for this is detected you will benefit from parmap parallelization.
+
+When parmap is enabled coccicheck will enable dynamic load balancing by using
+``--chunksize 1`` argument, this ensures we keep feeding threads with work
+one by one, so that we avoid the situation where most work gets done by only
+a few threads. With dynamic load balancing, if a thread finishes early we keep
+feeding it more work.
+
+When parmap is enabled, if an error occurs in Coccinelle, this error
+value is propagated back, the return value of the ``./scripts/coccicheck``
```

Comment:
here you're using a path for the script name, where previously you just used the script name.